src/Symfony/Component/DependencyInjection/Compiler/ResolveDefinitionTemplatesPass.php

```diff
@@ -95,33 +95,27 @@ private function doResolveDefinition(ChildDefinition $definition)
         if ($parentDef->isDeprecated()) {
             $def->setDeprecated(true, $parentDef->getDeprecationMessage('%service_id%'));
         }
-        $def->setFactory($parentDef->getFactory());
-        $def->setConfigurator($parentDef->getConfigurator());
-        $def->setFile($parentDef->getFile());
-        $def->setPublic($parentDef->isPublic());
-        $def->setLazy($parentDef->isLazy());
-        $def->setAutowired($parentDef->isAutowired());
 
-        self::mergeDefinition($def, $definition);
-
-        // merge autowiring types
-        foreach ($definition->getAutowiringTypes(false) as $autowiringType) {
-            $def->addAutowiringType($autowiringType);
+        $parentChanges = $parentDef->getChanges();
+        if (isset($parentChanges['factory'])) {
```

Comment:
I see one fragility here. 
What if the parent uses defaults?
```yml
services:
    _defaults:
        autowire: true

    parent: ~
```
```yml
services:
    Foo\Bar:
        parent: parent
```
Then `Foo\Bar` won't be autowired, right?

I think the defaults should always be tracked but we should allow to differentiate changes from defaults (maybe just internally, I see no need to do this in userland):
```php
if (isset($defaults['autowire'])) {
    $definition->setTrackChanges('defaults')
        ->setAutowired($defaults['autowire'])
        ->setTrackChanges(true);
}
```
and then use something like the following in the passes needing it:
```php
$definition->getChanges($withoutDefaults = true);
```